### ISSUE TITLE
Getting slower and slower the more entries

### ISSUE THREAD

**Issue by somatonic** [teppokoivula/ProcessChangelog#13]
Just as a info.

I have a live site where a lot of changes happen and the changelog would slow down loading editing screen and also the changelog screen took a long time to render. I talking about millions of entries. Opening a page took about 4-5 seconds. The changelog screen took about 20-30 seconds to load.

I deinstalled the module and now it's all back to normal.

Edit: I've seen there's a max date "expire" entries settings, which was off on that install I think. But anyway just wanted to put a note here.

**Comment by teppokoivula** [comment]
Thanks, @somatonic! I did try to make things as efficient as possible when first building this module, but I'll see if there's something I can optimise further. (Usually there is.)

That being said, there's always going to be an increasing speed impact as a database table gets bigger, so enabling a sensible max age setting is definitely recommended :)

**Comment by teppokoivula** [comment]
Thanks again. While testing with hundreds of thousands of rows I started seeing noticeable slowdowns on the Changelog page. This was a result of an obvious mistake by me, i.e. the module was calling certain method once per every single row in the database table :)

The issue with edit screen might've been separate issue. For the time being I've added a new option to the ProcessChangelogHooks module that allows one to disable the Changelog section on page editor. This is specifically recommended if there's no age limit defined for collected data.

I've also made some smaller improvements to generated SQL queries, which should hopefully speed things up a bit. It's quite likely that none of these are noticeable until you have at least tens or hundreds of thousands of rows.
